resourcemanager/pool/spool/spool.go

```diff
@@ -79,8 +79,23 @@ func (p *Pool) Tune(size int32) {
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	p.SetLastTuneTs(time.Now())
+	old := p.capacity
 	p.capacity = size
 	p.concurrencyMetrics.Set(float64(size))
+	if old == size {
+		return
+	}
+	if old < size {
+		_, task := p.taskManager.Overclock()
+		if task != nil {
+			p.running.Add(1)
+			p.run(func() {
+				runTask(task)
+			})
+		}
+		return
+	}
+	p.taskManager.Downclock()
```

Comment:
Another question is do we need to call `Downclock` async because it can be blocked by channel and what is worse, `p.mu` is locked when it happens.

Reply:
so It is the reason that it has a shard map to avoid frequent lock races. and His lock fine-grained is very small. it doesn't take too much time on it.

Reply:
I know. If I find that it is blocked from sending a msg to exitChan, I will skip it.